lib/repl.js

```diff
@@ -374,6 +374,17 @@ function REPLServer(prompt,
     };
   }
 
+  function prettifyStackTrace(err) {
+    if (err.stack) {
+      const regex = new RegExp(/\s+repl:/);
+      const stack = err.stack.split('\n');
+      const idx = stack.findIndex((frame) => regex.test(frame));
```

Comment:
This filters only a single frame.  Wouldn't it make more sense to do e.g.
```js
err.stack = err.stack.split('\n').filter((s) => !/ repl:/.test(s)).join('\n');
```

Reply:
This finds the index of ` repl:` and on line 383 below, the stack is spliced from that point, so all frames after it are removed. If I am reading it correctly, I believe your suggestion would only eliminate lines that begin with ` repl:`, but that's not all that we are looking for.  Here is some before/after:

```js
~/s/node (repl-prettify-stack-trace ☡+) node --version
v8.4.0
~/s/node (repl-prettify-stack-trace ☡+) node
> .load test/fixtures/repl-pretty-stack.js
> 'use strict';
'use strict'
> function a() {
...   b();
... }
undefined
> function b() {
...   c();
... }
undefined
> function c() {
...   throw new Error('Whoops!');
... }
undefined
> a();
Error: Whoops!
    at c (repl:2:9)
    at b (repl:2:3)
    at a (repl:2:3)
    at repl:1:1
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:120:20)
> ⏎                                                                                                                       ~/s/node (repl-prettify-stack-trace ☡+) ./node --version
v9.0.0-pre
~/s/node (repl-prettify-stack-trace ☡+) ./node
> .load test/fixtures/repl-pretty-stack.js
'use strict';
function a() {
  b();
  }
  function b() {
    c();
    }
    function c() {
      throw new Error('Whoops!');
      }
      a();

Error: Whoops!
    at c (repl:9:13)
    at b (repl:6:5)
    at a (repl:3:3)
>
```
As you can see, it takes care of stack frames including `REPLServer`, `ContextifyScript`, etc.